HACKING.md

```diff
@@ -1,5 +1,127 @@
+CI, but is close enough for many purposes. Put a symlink early in your path:
+```
+ln -s /usr/bin/podman /somewhere/early/in/my/PATH`
+```
+
+## Testing Overview
+
+There are two types of test:
+
+* *Unit tests:* run in seconds, no cluster needed, verify internal behavior - behind the API. New could *should* be reasonably well covered by unit tests but we don't have a formal requirement.
+* *E2E tests:* run in minutes to hour(s), requires a cluster. There are several ways to run these:
+   - *CI tests:* run automatically when you raise a PR in a controlled cluster environment. Can take hour(s) to get results.
+   - *Deployed tests:* Build and deploy CLO image to your own cluster, allows you to run tests selectively.
+   - *Local process:* requires a cluster with dependencies deployed (elasticsearch), but runs CLO as a *local process*. Faster and easier to debug.
+
+All PRs should pass unit tests. CI runs all tests automatically and won't merge a PR till they pass, so it is OK to create a PR and let CI run the e2e tests if you've done enough testing to be confident. This allows for human review and feedback while CI tests are running.
+
+To get a code review of unfinished work, create a PR with "[WIP]" at the start of the description. The CI system will not merge it until the "[WIP]" is removed and all tests are passing.
+
+## Setting up a test cluster
+
+You can use a real cluster (for example an AWS cluster) or create a virtual cluster on your development box with [Code Ready Containers](https://developers.redhat.com/products/codeready-containers/download)
+
+If you use CRC, you may need to start with more memory than the default, e.g.:
+```
+crc start -m 12288
+```
+
+Log in as a user with the role `cluster-admin` - user `kubeadmin` is often predefined with this role.
+
+You can `export KUBECONFIG=/path/to/cluster/config` or copy/add your config to `$HOME/.kube/config`. Depending on how you set it up, you may also need to `oc login` to your cluster and add pull secrets to your credentials.
+
+## Running tests
+
+Unit tests can be run without a cluster:
+* `make`: Run unit tests, lint and format code. Do this before submitting a PR.
+* `make test-unit`: Just run unit tests.
+
+To build, deploy and test the CLO image in your own cluster:
+* `make deploy`: Build CLO image and deploy to cluster with elasticsearch.
+* `make undeploy`: Undo `make deploy`.
+
+To run CLO as a local process:
+* `make run`: Run CLO as a local process. Does not require `make deploy`.
+* `make debug`: Run CLO in the `dlv` debugger.
+
+After running `make deploy`, `make run` or `make debug`, you can run e2e tests:
+* `make test-cluster` Run all e2e tests against the running CLO.
+* `go run ./test/e2e/something`: Run tests individually, in a debugger or whatever.
+
+To do a complete deploy, test, cleanup cycle exactly as CI does (this must  start from a clean cluster e.g. after `make undeploy`):
+* `make test-olm-e2e`: Run e2e tests the same way CI does
```

Comment:
this requires some env vars to be set if you want it to actually run your code and not just the latest images on api.ci, right?

Reply:
Yes, I think this is documented in the Makefile itself, if not we should add it in a future PR.